src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -408,47 +395,35 @@ var ReactCompositeComponentMixin = {
     updateQueue
   ) {
     var Component = this._currentElement.type;
-    var instanceOrElement;
+
+    var debugID = 0;
+    if (__DEV__) {
+      debugID = this._debugID;
+    }
+
     if (doConstruct) {
-      if (__DEV__) {
-        if (this._debugID !== 0) {
-          ReactInstrumentation.debugTool.onBeginLifeCycleTimer(
-            this._debugID,
-            'ctor'
-          );
-        }
-      }
-      instanceOrElement = new Component(publicProps, publicContext, updateQueue);
-      if (__DEV__) {
-        if (this._debugID !== 0) {
-          ReactInstrumentation.debugTool.onEndLifeCycleTimer(
-            this._debugID,
-            'ctor'
-          );
-        }
+      if (__DEV__ && debugID !== 0) {
+        return measureLifeCyclePerf(
+          () => new Component(publicProps, publicContext, updateQueue),
+          debugID,
+          'ctor'
```

Comment:
It's a reserved name in JS so it won't work as object key. We use lifecycle names as keys in the ReactPerf detailed output. We could prefix them but this makes output more convoluted, and `constructor` is the only problematic name.